code/__HELPERS/game.dm

```diff
@@ -433,22 +433,55 @@
 
 	return result
 
-/proc/pollCandidatesForMob(Question, jobbanType, be_special_flag = 0, poll_time = 300, mob/M, ignore_category = null)
-	var/list/L = pollGhostCandidates(Question, jobbanType, be_special_flag, poll_time, ignore_category)
-	if(!M || QDELETED(M) || !M.loc)
+/**
+ * Returns a list of ghosts that are eligible to take over and wish to be considered for a mob.
+ *
+ * Arguments:
+ * * question - Question to show players as part of poll
+ * * jobban_type - Type of jobban to use to filter out potential candidates.
+ * * be_special_flag - Unknown/needs further documentation.
+ * * poll_time - Length of time in deciseconds that the poll input box exists before closing.
+ * * target_mob - The mob that is being polled for.
+ * * ignore_category - Unknown/needs further documentation.
+ */
+/proc/poll_candidates_for_mob(question, jobban_type, be_special_flag = 0, poll_time = 300, mob/target_mob, ignore_category = null)
+	var/static/list/mob/currently_polling_mobs = list()
+
+	if(currently_polling_mobs.Find(target_mob))
 		return list()
-	return L
-
-/proc/pollCandidatesForMobs(Question, jobbanType, be_special_flag = 0, poll_time = 300, list/mobs, ignore_category = null)
-	var/list/L = pollGhostCandidates(Question, jobbanType, be_special_flag, poll_time, ignore_category)
-	var/i=1
-	for(var/v in mobs)
-		var/atom/A = v
-		if(!A || QDELETED(A) || !A.loc)
-			mobs.Cut(i,i+1)
-		else
-			++i
-	return L
+
+	currently_polling_mobs += target_mob
+
+	var/list/possible_candidates = pollGhostCandidates(question, jobban_type, be_special_flag, poll_time, ignore_category)
+
+	currently_polling_mobs -= target_mob
+	if(!target_mob || QDELETED(target_mob) || !target_mob.loc)
+		return list()
+
+	return possible_candidates
+
+/**
+ * Returns a list of ghosts that are eligible to take over and wish to be considered for a mob.
+ *
+ * Arguments:
+ * * question - Question to show players as part of poll
+ * * jobban_type - Type of jobban to use to filter out potential candidates.
+ * * be_special_flag - Unknown/needs further documentation.
+ * * poll_time - Length of time in deciseconds that the poll input box exists before closing.
+ * * mobs - The list of mobs being polled for. This list is mutated and invalid mobs are removed from it before the proc returns.
+ * * ignore_category - Unknown/needs further documentation.
+ */
+/proc/poll_candidates_for_mobs(question, jobban_type, be_special_flag = 0, poll_time = 300, list/mobs, ignore_category = null)
```

Comment:
```suggestion
/proc/poll_candidates_for_mobs(question, jobban_type, be_special_flag = 0, poll_time = 30 SECONDS, list/mobs, ignore_category = null)
```

Reply:
You might also want to do this for all of the cases where you call this proc, because they use values such as `75` instead of `7.5 SECONDS`